Update README.md (#26)
checkout/checkout-3ds-sdk-ios: README.md

5. Meets requirements from **EMVCo and the PCI Security Standards Council**, specifically set for 3DS SDKs, so you can be sure it is interoperable with card issuers and that your customers’ sensitive data stays secure.

## Minimum Requirements

> The 3DS SDK for iOS requires Xcode 13.1 and above with Swift version 5.6.2 and above, and supports apps targeting iOS 11.0 and above. It also supports Objective-C. There are two ways to integrate into our 3DS SDK.

Reviewer: Deployment target changed from iOS 12 -> iOS 11 ?

## Getting Started

First, integrate the 3DS SDK into your app.

### Integrating with CocoaPods
[CocoaPods](http://cocoapods.org) is the traditional dependency manager for Apple projects. We still support it, but we're not always able to validate all its peculiar ways.

Make sure cocoapods is installed on your machine by running
```bash
$ pod --version
```
Any version newer than **1.10.0** is a good sign. If not installed, or unsupported, follow [Cocoapods Getting Started](https://guides.cocoapods.org/using/getting-started.html)

Once Cocoapods of a valid version is on your machine, to integrate Frames into your Xcode project, update your `Podfile`:
```ruby
platform :ios, '12.0'
use_frameworks!

target '<Your Target Name>' do

>   pod 'JOSESwift', '2.2.1', :inhibit_warnings => true

Reviewer 2: I'm not aware of the challenges or the reasons that we require the merchants to include this into their `Podfile`s but could this have been a dependency of our SDK?

    appURL: URL(string: "myapp://my-app-url")!
)

let authenticationParameters = AuthenticationParameters(
    sessionID: sessionID,
    sessionSecret: sessionSecret,
    scheme: scheme)

checkout3DS.authenticate(authenticationParameters: authenticationParameters) { error in
    if let error = error {
        // Handle error.
    } else {
        // Continue with payment.
    }
}
```
👉&nbsp;&nbsp;[See the Integration Guide with Code Examples](https://www.checkout.com/docs/risk-management/3d-secure/sessions/non-hosted-sessions/3d-secure-mobile-sdks) for full details.

📚&nbsp;&nbsp;[Read the reference documentation](https://checkout.github.io/checkout-mobile-docs/checkout-3ds-sdk-ios/index.html)

### Integrating with any 3DS provider
Standalone 3DS allows our SDKs to be used with any Authentication provider, be it Checkout.com or otherwise. This is a higher touch integration that breaks up the 3DS flow more granularly. In order to integrate with the Standalone 3DS service:

1. Initialize the SDK with your preferred user interface options using our `uiCustomization` object.
2. Create `transaction` object 
3. Get `authenticationRequestParameters` for the AReq

> 4. if a challenge is mandated from the Authentication response from your 3DS server then call the `doChallenge` method to render the challenge however, if the challenge is not mandated by the ACS then it would have triggered a frictionless 3DS flow. 

Reviewer 2: ```suggestion
4. If a challenge is mandated from the Authentication response from your 3DS server then call the `doChallenge` method to render the challenge however, if the challenge is not mandated by the ACS then it would have triggered a frictionless 3DS flow. 
```

#### Diagram
Here is a useful diagram that highlights the end-to-end 3DS flow using our `standalone3DSService`.
![E2E Standalone SDK Flow](https://user-images.githubusercontent.com/102961713/226956636-d39b7bff-9fb8-4701-a3bc-86d932b306f0.jpg)

#### Code snippet 

1. Creates an instance of `ThreeDS2Service` through which the 3DS Requestor App can create a transaction object to get the `authenticationRequestParameters` that are required to perform a challenge:

```swift
   private var transaction: Transaction?
   private var standalone3DSService: ThreeDS2Service?
````

2. Initialize the SDK with your preferred user interface options:

> - `scheme` can be set only as `visa` and `mastercard` in lower case string.

Reviewer 2: ```suggestion
- `scheme` can be set only as `visa` and `mastercard` in lowercase string.
```

                                                                              directoryServerPublicKey: <ds_public>,
                                                                              directoryServerRootCertificates: [<caPublic>])
   let configParameters = ThreeDS2ServiceConfiguration.ConfigParameters(directoryServerData: directoryServerData,
                                                                        messageVersion: <messageVersion>,
                                                                        scheme: <scheme>)
    let serviceConfiguration = ThreeDS2ServiceConfiguration(configParameters: configParameters)
      
    self.standalone3DSService = try Standalone3DSService.initialize(with: serviceConfiguration)
   }  catch let error {
      // handle failure scenario
   }
```

3. Create `transaction` using created `ThreeDS2Service` object:

```swift
  self.transaction = self.standalone3DSService?.createTransaction()
```

4. Retrieve the `authenticationRequestParameters`:
```swift
 // get Authentication Request parameters
 self.transaction?.getAuthenticationRequestParameters { result in
     switch result {
     case .success(let params):

>         //Make an Authentication Request to your 3DS Server

Reviewer 2: ```suggestion
        // Make an Authentication Request to your 3DS Server
```

     case .failure(let error):
        // handle failure scenario
      }
}

```
5. Handle the `doChallenge` flow: 

- If the Authentication response that is returned indicates that the Challenge Flow must be applied, the 3DS Requestor App calls the `doChallenge` method with the required input `ChallengeParameters`. The `doChallenge` method initiates the challenge process.
- The `doChallenge` function now returns a `ChallengeResult` object with `transactionStatus` and `sdkTransactionID`.
```swift
let params = ChallengeParameters(threeDSServerTransactionID: response.transactionId,
                                 acsTransactionID: response.acs.transactionId,
                                 acsRefNumber: response.acs.referenceNumber,
                                 acsSignedContent: response.acs.signedContent)
transaction?.doChallenge(challengeParameters: params, completion: { [weak self]  result in
     switch result {
      case .success(let result):
          // handle success scenario
      case .failure(let error):
         // handle failure scenario
       }
      // call close and cleanUp methods after challenge flow is completed. 
      self?.transaction?.close()

>       self?.standalone3DSService?.cleanUp()

Reviewer 2: re, I'm not aware of the actual reason so a bit of an uneducated guess here: Why don't we have a strong reference to `self` in this closure to avoid, for example, `close()` to be called but `self` being deallocated and `cleanUp()` is not being called?
